### ISSUE TITLE
Multiple preset download paths

### ISSUE THREAD

**Issue by drws** [transmission/transmission#6044]
### What is the issue?

Currently only the main download path/directory is preset in Transmission settings. However, in practice more than one download location is often used. Transmission could support presetting multiple paths and offer a selection between them in the _Add Torrent_ dialog window (currently probably wrongly titled _Torrent Options_ - see #6043).

Any history-based solution is cumbersome. The list needs to be static, like the desktop (XDG) folders are for example.

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.3

**Comment by killemov** [comment]
I agree and implemented it in [Shift](https://github.com/killemov/Shift#paths-paths-paths). It uses scraping for an initial list, is fully editable and has some more goodies. This has been discussed in the past and some intrinsic motivation for implementing this appears to be lacking.

**Comment by mikedld** [comment]
I'm assuming you're using GTK 4 build, as with GTK 3 a standard (and now deprecated) FileChooserButton widget is being used which supports this out of the box.

**Comment by killemov** [comment]
> I'm assuming you're using GTK 4 build, as with GTK 3 a standard (and now deprecated) FileChooserButton widget is being used which supports this out of the box.

I'm assuming your assumption is wrong because this isn't that big of a problem with native clients. @drws listed transmission-daemon as the affected application so I was assuming the web UI instead.

**Comment by drws** [comment]
It seems you're both missing the actual point.

Sorry for _transmission-daemon_ confusion. What I wanted to say is that the feature should be supported in core Transmission code, regardless of UI. So the functionality would be available in CLI, WebUI, GTK3, GTK4, Qt and all the future ones. Should've explained this before.

The point is that Transmission should support multiple preset download paths in the config file, one of which is the default one. Path presetting should be core (not UI) functionality and the concept of path selection incorporated in all the relevant places (GUIs for example).

**Comment by mikedld** [comment]
This feels like a part of a bunch of other requests like #79, #972, #3813, #6129. Having an overly narrow view of the issue will result in more maintenance and backward compatibility headache down the road (e.g. #1298).

**Comment by killemov** [comment]
@mikedld I disagree. This specific issue is just about having a set of paths available for selection when adding or moving a torrent. And as I stated previously this is a bigger problem for transmission-daemon+web-UI as the RPC does not have a way to select any path on the server so you have to enter paths manually. So does my solution for [Shift](https://github.com/killemov/Shift), but you only have to enter a path once or you can scrape the paths from existing torrents. No future compatibility headaches, just ... [Paths! Paths! Paths!](https://github.com/killemov/Shift#paths-paths-paths)

**Comment by jpovixwm** [comment]
@killemov the point of #6426 is that you can call `torrent-add` today with `download-dir: "C:/ThisDirDoesNotExist"` and the directory will be created if it doesn't already exist, but if you call `free-space` with `path: "C:/ThisDirDoesNotExist"` beforehand, you will get an error response.
It's proposed that `free-space` should instead find out where `torrent-add` would create the new directory, and return the remaining space available on the corresponding drive, in this case `C:/`.

I can't really see any clear connection between this issue and #6426.

**Comment by drws** [comment]
> This feels like a part of a bunch of other requests like #79, #972, #3813, #6129. Having an overly narrow view of the issue will result in more maintenance and backward compatibility headache down the road (e.g. #1298).

All these feature requests are about a more complicated grouping functionality and related features. Transmission already supports custom download paths, this is just about a simple ability to have some presets to an already-present functionality.

**Comment by killemov** [comment]
@jpovixwm 
> I can't really see any clear connection between this issue and #6426.

The connection is that a set of preset paths is missing either for selection or for free space queries. So the same set should be multi-purpose.
